config/routes.rb

```diff
@@ -1,3 +1,17 @@
 Rails.application.routes.draw do
+  resources :posts
+  resources :blogs do
+    collection do
+      get 'manage'
+    end
+    member do
+      get 'posts'
+      get 'post_details/:post_id', action: :post_details, as: :post_details
+    end
```

Comment:
Any reason not to use nested resources for blogs -> posts here, rather than having a `posts` method and a `post_details` method on the blogs controller?  Feels a bit weird to have methods relating to posts on a blogs controller!